hq/app/aws/ec2/EC2.scala

```diff
@@ -120,4 +122,26 @@ object EC2 {
         UnknownUsage(ni.getDescription, ni.getNetworkInterfaceId)
       )
   }
+
+  private[ec2] def addVpcName (flaggedSgs : List[SGOpenPortsDetail], vpcs: Map[String, Vpc]): List[SGOpenPortsDetail] = {
+    for {
+      sgs <- flaggedSgs
+      vpc <- vpcs.collectFirst{ case (id, v) if id == sgs.vpcId => v }
+      vpcN  = vpc.getTags.asScala.collectFirst{ case tag if tag.getKey == "Name" => tag.getValue }
+      sgsUpdated = sgs.copy(vpcName = vpcN)
+    } yield sgsUpdated
+  }
+
+  private[ec2] def getVpcs(account: AwsAccount, flaggedSgs: List[SGOpenPortsDetail])(vpcsDetailsF: (AmazonEC2Async, List[String]) => Attempt[Map[String, Vpc]])(implicit ec: ExecutionContext): Attempt[Map[String, Vpc]] = {
+    Attempt.traverse(flaggedSgs.map(_.region).distinct) { region =>
+      val ec2Client = client(account, Regions.fromName(region) )
+      val vpcIds = flaggedSgs.filter(_.region == region).map(_.vpcId)
+      vpcsDetailsF(ec2Client, vpcIds)
+    }.map( _.fold(Map.empty)(_ ++ _) )
+  }
+
+  private def getVpcsDetails (client: AmazonEC2Async, vpcIds: List[String])(implicit ec: ExecutionContext): Attempt[Map[String, Vpc]] = {
+    val request = new DescribeVpcsRequest().withVpcIds(vpcIds.asJava)
+    handleAWSErrs(awsToScala(client.describeVpcsAsync)(request)).map( _.getVpcs.asScala.map(vpc => vpc.getVpcId -> vpc).toMap)
```

Comment:
This looks a little complex for a single line, is it possible to break it up slightly?